src/utils/api/api.ts

```diff
@@ -171,14 +172,17 @@ export const removeReadonlyFields = <T extends z.ZodRawShape, TUnwrap extends (k
   shape: T,
   unwrap?: TUnwrap
 ) => {
-  const entries = Object.entries(shape).flatMap(([k, v]) => {
```

Comment:
Reworked this trying to avoid any incompatibility issues with flatMap but that seemed not to be the case